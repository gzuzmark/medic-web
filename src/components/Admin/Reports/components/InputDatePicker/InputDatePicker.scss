.InputDatePicker {
  &--error {
    .DateInput_input {
      border-bottom: 1px solid #ee5c7d;
      color: #ee5c7d;
    }

  }
}