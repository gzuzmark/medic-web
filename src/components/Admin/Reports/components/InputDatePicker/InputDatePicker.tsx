import * as moment from "moment";
import * as React from "react";
import DatePickerBase from "../../../../../common/DatePickerBase/DatePickerBase";
import './InputDatePicker.scss';

interface IStateInputDatePicker {
    focus: boolean;
}

interface IPropsInputDatePicker {
    id: string;
    configDate?: object;
    date: Date;
    error?: boolean;
    updateState: (params: object) => void;
}

class InputDatePicker extends React.Component <IPropsInputDatePicker, IStateInputDatePicker> {
    public state: IStateInputDatePicker;
    constructor(props: IPropsInputDatePicker) {
        super(props);
        this.state = {
            focus: false
        };
        this.onDateChange = this.onDateChange.bind(this);
        this.onDateFocusChange = this.onDateFocusChange.bind(this);
    }

    public render() {
        return (
            <div className={`InputDatePicker ${this.props.error ? 'InputDatePicker--error' : ''}`}>
                <DatePickerBase
                    id={this.props.id}
                    date={moment(this.props.date)}
                    focus={this.state.focus}
                    onDateChange={this.onDateChange}
                    onDateFocusChange={this.onDateFocusChange}
                    configs={this.props.configDate} />
            </div>
        )
    }

    private onDateChange(date: moment.Moment) {
        this.props.updateState({[this.props.id]: date.toDate()})

    }

    private onDateFocusChange(focusedInput: any) {
        const focus = focusedInput ? focusedInput.focused : false;
        this.setState({focus});
    }


}

export default InputDatePicker;
