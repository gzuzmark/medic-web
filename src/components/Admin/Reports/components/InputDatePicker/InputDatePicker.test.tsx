import { shallow } from "enzyme";
import * as React from 'react';
import InputDatePicker from "./InputDatePicker";

describe('InputDatePicker Test',() => {
    let props: any;
    let mountedInputDatePicker: any;
    const getComponent = () => {
        if (!mountedInputDatePicker) {
            mountedInputDatePicker = shallow(
                <InputDatePicker {...props} />
            );
        }
        return mountedInputDatePicker;
    };

    beforeEach(() => {
        props = {
            date: new Date("2018-02-02"),
            id: 'demoKey',
            updateState: (params: object) => void(0)
        };
        mountedInputDatePicker = undefined;
    });

    it("render: render InputDatePicker", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });
});