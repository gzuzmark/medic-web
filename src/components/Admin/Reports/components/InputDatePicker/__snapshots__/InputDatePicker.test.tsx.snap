// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDatePicker Test render: render InputDatePicker 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputDatePicker
    date={2018-02-02T00:00:00.000Z}
    id="demoKey"
    updateState={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <DatePickerBase
        configs={undefined}
        date={"2018-02-02T00:00:00.000Z"}
        focus={false}
        id="demoKey"
        onDateChange={[Function]}
        onDateFocusChange={[Function]}
      />,
      "className": "InputDatePicker ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "configs": undefined,
        "date": "2018-02-02T00:00:00.000Z",
        "focus": false,
        "id": "demoKey",
        "onDateChange": [Function],
        "onDateFocusChange": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <DatePickerBase
          configs={undefined}
          date={"2018-02-02T00:00:00.000Z"}
          focus={false}
          id="demoKey"
          onDateChange={[Function]}
          onDateFocusChange={[Function]}
        />,
        "className": "InputDatePicker ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "configs": undefined,
          "date": "2018-02-02T00:00:00.000Z",
          "focus": false,
          "id": "demoKey",
          "onDateChange": [Function],
          "onDateFocusChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
